File: api/cas-mgmt-api-dashboard/src/main/java/org/apereo/cas/mgmt/domain/Cache.java
package org.apereo.cas.mgmt.domain;

import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Map;

/**
 * Claas used to serialize Cache data to client.
 *
 * @author Travis Schmidt
 * @since 6.1.0
 */
@Data
@NoArgsConstructor
public class Cache {

    /**
     * Health status of the cache.
     */
    private String status;

    /**
     * Cache details.
     */
    private Details details;

    @Data
    @NoArgsConstructor
    public static class Details {
        /**
         * Set of maps that are part of the cache and their details.
         */
        private Map<String, MapDetails> maps;
    }

    @Data
    @NoArgsConstructor
    public static class MapDetails {
        /**
         * Memory used by the cache map.
         */
        private Long memory;

        /**
         * Number of items in the cache map.
         */
        private Long size;

        /**
         * Total capacity of the cache map.
         */
        private Long capacity;
    }

}

File: webapp/cas-mgmt-webapp-workspace/projects/dashboard/src/app/domain/cache.ts
export class MapDetails {
  name: string;
  memory: number;
  size: number;
  capacity: number;
}

export class Details {
  name: string;
  sessionCount: number;
  ticketCount: number;
  message: string;
  maps: Map<string, MapDetails> = new Map();
}

export class Cache {
  status: string;
  details: Details = new Details();
}
